File: 2525-CountNumberOfDistinctIntegersAfterReverseOperations/2525-CountNumberOfDistinctIntegersAfterReverseOperations.py
# Last updated: 04/04/2026, 23:30:07
class Solution(object):
    def countDistinctIntegers(self, nums):
        distinct_numbers = set()
        for num in nums:
            distinct_numbers.add(num)
            reversed_num = int(str(num)[::-1])
            distinct_numbers.add(reversed_num)
        return len(distinct_numbers)

File: 2525-CountNumberOfDistinctIntegersAfterReverseOperations/2525-CountNumberOfDistinctIntegersAfterReverseOperations.java
// Last updated: 04/04/2026, 23:30:10
class Solution {
    public int reverseNumber(int num) {
        int reversed = 0;
        while (num > 0) {
            reversed = reversed * 10 + num % 10;
            num /= 10;
        }
        return reversed;
    }    
    public int countDistinctIntegers(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
            set.add(reverseNumber(num));
        }
        return set.size();
    }
}